src/rgw/rgw_rest.h

```diff
@@ -816,3 +816,5 @@ extern int dump_body(struct req_state* s, const char* buf, size_t len);
 extern int dump_body(struct req_state* s, /* const */ ceph::buffer::list& bl);
 extern int dump_body(struct req_state* s, const std::string& str);
 extern int recv_body(struct req_state* s, char* buf, size_t max);
+
+std::string lowercase_underscore_http_attr(const std::string& orig);
```

Comment:
So apparently this function is already implemented in src/rgw_rest.cc, although not declared `static`. Cool!